From: Folia Compatibility Patch
Subject: [PATCH 0006] Visuals-FancyMessage: Replace dispatchCommand logic with ComponentSerializer.parse()

AE PATCH REASON: FancyMessage uses legacy JSON conversion incompatible with modern Adventure API
AE PATCH FIX: Replace dispatchCommand logic with ComponentSerializer.parse()
AE PATCH RISK: LOW
AE PATCH EXCLUDES: net/advancedplugins/ae/impl/utils/fanciful/FancyMessage.class, net/advancedplugins/ae/utils/fanciful/FancyMessage.class

---
diff --git a/net/advancedplugins/ae/impl/utils/fanciful/FancyMessage.java b/net/advancedplugins/ae/impl/utils/fanciful/FancyMessage.java
index ef1877d..44b65fe 100644
--- a/net/advancedplugins/ae/impl/utils/fanciful/FancyMessage.java
+++ b/net/advancedplugins/ae/impl/utils/fanciful/FancyMessage.java
@@ -357,7 +357,7 @@ public class FancyMessage implements JsonRepresentedObject, Cloneable, Iterable<
       if (!(sender instanceof Player player)) {
          sender.sendMessage(this.toOldMessageFormat());
       } else {
-         Bukkit.dispatchCommand(Bukkit.getConsoleSender(), "tellraw " + player.getName() + " " + jsonString);
+         player.sendMessage(net.kyori.adventure.text.serializer.gson.GsonComponentSerializer.gson().deserialize(jsonString));
       }
    }
 

diff --git a/net/advancedplugins/ae/utils/fanciful/FancyMessage.java b/net/advancedplugins/ae/utils/fanciful/FancyMessage.java
index 2b15cd3..fdf083f 100644
--- a/net/advancedplugins/ae/utils/fanciful/FancyMessage.java
+++ b/net/advancedplugins/ae/utils/fanciful/FancyMessage.java
@@ -363,7 +363,7 @@ public class FancyMessage implements JsonRepresentedObject, Cloneable, Iterable<
       if (!(sender instanceof Player player)) {
          sender.sendMessage(this.toOldMessageFormat());
       } else {
-         Bukkit.dispatchCommand(Bukkit.getConsoleSender(), "tellraw " + player.getName() + " " + jsonString);
+         player.sendMessage(net.kyori.adventure.text.serializer.gson.GsonComponentSerializer.gson().deserialize(jsonString));
       }
    }
 

